### ISSUE TITLE
Update to use apiCheck as a function

### ISSUE THREAD

**Issue by benoror** [formly-js/angular-formly-templates-bootstrap#40]
As apiCheck as an object [has been deprecated](https://github.com/formly-js/angular-formly/blob/6.20.0/other/ERRORS_AND_WARNINGS.md#apicheck-as-an-object-deprecated) (see: https://github.com/formly-js/angular-formly/issues/334), custom types and wrappers [should be updated accordingly](https://github.com/formly-js/angular-formly-templates-bootstrap/search?utf8=%E2%9C%93&q=apicheck).

**Comment by benoror** [comment]
I'm willing to work on this chore, but I'm not sure how to proceed, as there's [no updated docs regarding it](http://docs.angular-formly.com/v6.18.0/docs/custom-templates#section-formlyconfig-settype-options). Would this be correct?:

###### Example: src/types/checkbox.js

```
export default ngModule => {
  ngModule.config(addCheckboxType);

  function addCheckboxType(formlyConfigProvider, formlyBootstrapApiCheck) {
    const c = formlyBootstrapApiCheck;
    formlyConfigProvider.setType({
      name: 'checkbox',
      template: require('./checkbox.html'),
      wrapper: ['bootstrapHasError'],
      apiCheck: function() {
        templateOptions: c.shape({
          label: c.string
        })
      },
      apiCheckInstance: c
    });
  }
};
```

**Comment by kentcdodds** [comment]
Yes, totally forgot to update this. You almost have it. The only difference is you no longer need to specify that `templateOptions` is a shape because that is assumed. So it'd look more like this:

```javascript
{
  apiCheck(check) {
    return {
      templateOptions: {label: check.string}
    };
  }
}
```

**Comment by benoror** [comment]
Good, gonna send a PR asap

**Comment by kentcdodds** [comment]
Remember, this is built with es6 :-) So you could go crazy and do this:

```javascript
apiCheck: check => ({templateOptions: {label: check.string}})
```

but that may be pushing it a bit ;-)

**Comment by kentcdodds** [comment]
Note, this will technically be a breaking change because this new format isn't supported by older versions of angular-formly and we'll have to update [the peerDep](https://github.com/formly-js/angular-formly-templates-bootstrap/blob/master/package.json#L37) to be: `>=6.20.0`. Therefore, this will push angular-formly-templates-bootstrap to `6.0.0`. Not bad, just wanna follow semver :-)

**Comment by kentcdodds** [comment]
Don't you worry about doing that for your PR though. I'll take care of that.

**Comment by kentcdodds** [comment]
Released as `6.0.0`! Thanks~
